### PR TITLE
Display the events joined by the user as a separate section of the list of events.

### PR DESCRIPTION
**Pull request by AlexandraImbrisca**
**Can't be merged before #29.**

In the current state, the list of events associated with a group only contains the events not joined by the user yet. This pull request introduces a **separation of the events in two different containers based on the user join status**. 

The main changes are:
- The loadGroupEvents function was generalised to handle both of the cases.
- The doGet method of the JoinedEventsServlet was modified such that it will retrieve all the events joined from a given group if specified. Otherwise, it will retrieve all the events joined by the user from all the groups.
- The DatastoreAccess.java class contains a new method to obtain only the events joined by the user from a given group.

### CODE REVIEW

**Review by tatiana-s — APPROVED** [review]

File: src/main/webapp/groups-script.js [hunk 1]
```diff
@@ -222,14 +222,21 @@ function joinGroup(groupId) {
 }
 
 /**
- * Fetches the events associated with the group from the server.
- * @param {Object} groupId The group id.
+ * Fetches events (from the group specified by the groupId) from the servlet
+ * and displays them in the given container. The function assumes the user has
+ * joined all the events or none at all.
```
Comment (nataliakrylova) [inline src/main/webapp/groups-script.js:227]: What does "The function assumes the user has joined all the events or none at all." mean? Is this a function which fetches a list of events which user has joined or a list of events which user has not joined?

Comment (AlexandraImbrisca) [reply src/main/webapp/groups-script.js:227]: "The function assumes the user has joined all the events or none at all." means that we can use the function for getting a list of the events which the user has joined and getting a list of the events which the user has not joined, but we can't use it for getting a mixture of these lists (e.g. some events joined by the user + some events not joined by the user). The reason for which I imposed this restriction is because we use different servlets for each category (joined/not joined) but we are not currently providing any servlet to get both joined and not joined events. I just wanted to emphasize this difference. Do you think a more detailed explanation might be useful?

Comment (nataliakrylova) [reply src/main/webapp/groups-script.js:227]: I understand now! Maybe rephrase it slightly to say that the function fetches/displays either all joined or all non-joined events? I was just slightly confused by "The function assumes" part because it sounded like a precondition I have to meet.

**Review by nataliakrylova — APPROVED** [review]
